File: app/src/main/java/com/team2052/frckrawler/activities/ServerLogActivity.java
package com.team2052.frckrawler.activities;

import android.os.Bundle;
import android.text.format.DateFormat;
import android.view.MenuItem;
import android.widget.ListView;

import com.team2052.frckrawler.R;
import com.team2052.frckrawler.adapters.ListViewAdapter;
import com.team2052.frckrawler.listitems.ListItem;
import com.team2052.frckrawler.listitems.elements.KeyValueListElement;
import com.team2052.frckrawler.theme.Themes;

import rx.Observable;

public class ServerLogActivity extends DatabaseActivity {

    private ListView mListView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        setTheme(Themes.getCurrentTheme(this).getTheme());
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_list_view);
        mListView = (ListView) findViewById(R.id.list);

        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        rxDbManager.getServerLog()
                .flatMap(Observable::from)
                .map(entry -> {
                    String time = DateFormat.getTimeFormat(this).format(entry.getTime());
                    return (ListItem) new KeyValueListElement(time, entry.getMessage());
                })
                .toList()
                .subscribe(listItems -> {
                    ListViewAdapter listViewAdapter = new ListViewAdapter(this, listItems);
                    mListView.setAdapter(listViewAdapter);
                });
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    public void inject() {
        getComponent().inject(this);
    }
}

File: app/src/main/java/com/team2052/frckrawler/activities/ServerLogActivity.kt
package com.team2052.frckrawler.activities

import android.os.Bundle
import android.text.format.DateFormat
import android.view.MenuItem
import android.view.View
import android.widget.ListView
import com.team2052.frckrawler.R
import com.team2052.frckrawler.adapters.ListViewAdapter
import com.team2052.frckrawler.adapters.items.ListItem
import com.team2052.frckrawler.adapters.items.elements.KeyValueListElement
import com.team2052.frckrawler.core.data.models.ServerLogEntry
import rx.Observable

class ServerLogActivity : DatabaseActivity() {

    private var mListView: ListView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_list_view)
        mListView = findViewById<View>(R.id.list) as ListView

        supportActionBar!!.setDisplayHomeAsUpEnabled(true)

        rxDbManager.serverLog
                .flatMap<ServerLogEntry> { Observable.from(it) }
                .map { entry ->
                    val time = DateFormat.getMediumDateFormat(this).format(entry.time)
                    KeyValueListElement(time, entry.message)
                }
                .toList()
                .subscribe { listItems ->
                    val listViewAdapter = ListViewAdapter(this, listItems as List<ListItem>?)
                    mListView!!.adapter = listViewAdapter
                }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
        }
        return super.onOptionsItemSelected(item)
    }

    override fun inject() {
        component.inject(this)
    }
}
